src/reducers/ratesReducer.js

```diff
@@ -17,36 +17,77 @@
     with this program; if not, write to the Free Software Foundation, Inc.,
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
-import { UPDATE_RATES } from 'constants/ratesConstants';
 
-type Rates = {
-  [key: string]: {
-    [key: string]: number,
-  },
-};
+// constants
+import {
+  SET_CHAIN_RATES,
+  SET_FETCHING_RATES,
+  SET_RATES,
+} from 'constants/ratesConstants';
+
+// types
+import type { RatesPerChain, RatesBySymbol } from 'models/Rates';
+
 
 export type RatesReducerState = {
-  data: Rates,
-  isFetched: boolean,
+  data: RatesPerChain,
+  isFetching: boolean,
 };
 
-export type RatesReducerAction = {
-  type: string,
-  payload: any
-};
+export type SetFetchingRatesAction = {|
+  type: typeof SET_FETCHING_RATES,
+  payload: boolean,
+|};
+
+export type SetRatesAction = {|
+  type: typeof SET_RATES,
+  payload: RatesPerChain,
+|};
+
+export type SetChainRatesAction = {|
+  type: typeof SET_CHAIN_RATES,
+  payload: {
+    chain: string,
+    rates: RatesBySymbol,
+  },
+|};
+
+export type RatesReducerAction = SetFetchingRatesAction
+  | SetRatesAction
+  | SetChainRatesAction;
 
 export const initialState = {
-  data: {},
-  isFetched: false,
+  data: { ethereum: {} },
+  isFetching: false,
 };
 
 const ratesReducer = (
   state: RatesReducerState = initialState,
   action: RatesReducerAction,
 ): RatesReducerState => {
   switch (action.type) {
-    case UPDATE_RATES:
-      return { ...state, data: action.payload, isFetched: true };
+    case SET_RATES:
+      return {
+        ...state,
+        data: action.payload,
+      };
+
+    case SET_CHAIN_RATES:
+      const { chain, rates } = action.payload;
+      return {
+        ...state,
+        data: {
+          ...state.data,
+          [chain]: rates,
```

Comment:
The way in other code uses `SET_CHAIN_RATES` suggests that this action should apply new rates over the previous ones.  Also consider renaming to `UPDATE_CHAIN_RATES` to reflact that fact.

```suggestion
          [chain]: { ...state.data?.[chain], ...rates},
```